### PR TITLE
Stop: Add stop_after_exception_count.

### PR DESCRIPTION
**Pull request by kobybum**
Add the ability to stop after a specific exceptions occurred X times.

### CODE REVIEW

File: tenacity/__init__.py [hunk 1]
```diff
@@ -412,6 +414,8 @@ def __init__(self, retry_object, fn, args, kwargs):
 
         #: The number of the current attempt
         self.attempt_number = 1
+        #: Tracks exceptions raised by the method, by type
+        self.exception_counts = Counter()
```
Comment (jd) [inline tenacity/__init__.py]: There should not be any need to store the number of exceptions here. The `stop_after_exception_count` class can store them AFAIU.

Comment (kobybum) [reply tenacity/__init__.py]: You're right, I thought it would keep state between calls, updated

**Comment by kobybum** [comment]
Nevermind, it looks like the state is kept between invocations, resulting in unwanted behavior.
This is the reason I opened the PR, since the handler doesn't work without inside modification.
I'll upload a failing test, let me know what you think, but this prevents us from using tenacity in a few places

**Comment by jd** [comment]
Did you give up on this?

**Pull request closed without merging**